spec/lib/puppet/test_ca.rb

```diff
@@ -49,6 +49,12 @@ def revoke(cert)
       @crl.add_revoked(revoked)
     end
 
+    def generate_client(name, opts)
+      host_key = OpenSSL::PKey::RSA.new(1024)
+      csr = create_csr(name, host_key)
+      { private_key: host_key, csr: csr, cert: sign(csr, opts).content }
+    end
+
     def generate(name, opts)
```

Comment:
Does it maybe make sense to just replace this method with your new one that returns more info?

Reply:
sure let me see what that entails.

Reply:
That was easier than expected. Updated.